### PR TITLE
Fix iconColor not applying to svg icons

### LINKED ISSUE #888: Icon color not working correctly
### For SVGs:
Icon color is currently being finally set within the `renderSvg` call. There, it is pulling the color from `svgIconProps.color` which is the same as `iconProps.svgSource.color`. In actuality, color is being passed in at `iconProps.color`. So `svgIconProps.color` is always undefined and thus the color of the icon defaults to white or black depending on light/dark mode
By changing `svgIconProps.color` to `iconProps.color`, the color that users pass in gets recognized and works. However, in the call to make the Icon, if the user doesn't pass in a color, the color that gets passed to `renderSvg` is `theme.colors.buttonText`. (Minor change there, in Icon component creation, `props.color` should be `rest.color` as `rest` is the most updated passed in props at that point.) This works fine for default or ghost buttons but for primary and fab styled buttons this causes the icon to blend in with the background.
For experimental button, the icon is set with a `tintColor` style which tints it to the accurate color for each button. To pull this color within Icon, it should be `iconProps.style.tintColor`. However, if we pull this color, we do not pull the color that the user passes in so the user is unable to customize the color of the icon. Also, the non-experimental button does not set a tint color so all icons in those buttons will be white/black depending on light/dark mode.

### For raster image (PNG) icons:
`tintColor` accurately tints the icon to the correct color. Currently icons in the non-experimental button are not being tinted to any color so they appear as their natural base color. There is no need to change any code.

### For font icons:
Similar to SVG icons in the fact that changing `iconProps.color` to `iconProps.style.tintColor` in `renderFontIcon` tints the icon to the appropriate color but ignores custom colors. Currently the icons blend into the background for fab and primary styled buttons.

### The fix:
The icon should check if the user passed in a color, if so, use that color. If the user did not pass in a color, check if a tint color exists. If it does, use the tint color. Finally if the tint color does not exist, then default to black/white depending on light/dark mode.
Also, adding a `tintColor` style prop to the icon in the non-experimental button may be nice so that the icons match the button theme if a custom color is not passed in.

### Current coloring:
Experimental button
Screen Shot 2021-08-11 at 2 40 17 PM
Non-experimental button
Screen Shot 2021-08-11 at 2 41 09 PM

### PR DESCRIPTION
**Pull request by rurikoaraki**
Fixes #888 

### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [x] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

When I made my ColorValue changes, I split out the color prop for the SVG icon into a separate prop, since RN SVG is still on an older version of RN and doesn't support ColorValue yet, only taking string. However I did this without really understanding how the token/slot/styling system worked and broke things for other components.

This change removes the color prop on the SvgIconProps and makes the svg icon use the color prop on IconProps as its base instead. I implemented a downgradeColor function which falls back to an appropriate color if the color is not a string.

I'm open to suggestions on better ways to gracefully handle the downgrade, but this won't really be fixed until RN SVG can handle ColorValues.

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| image | image |

These are actually both supposed to be red, but the customize for the iconColor token isn't working on the Button. Still looking into that.

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/experimental/Icon/src/Icon.tsx [hunk 1]
```diff
@@ -106,3 +101,11 @@ export const Icon = stagedComponent((props: IconProps) => {
...
 
 export default Icon;
+
+function downgradeColor(color: ColorValue): string {
+  if (typeof color === 'string') {
+    return color as string;
+  }
+
+  return getCurrentAppearance(useFluentTheme().host.appearance, 'light') === 'dark' ? '#FFFFFF' : '#000000';
+}
```
Comment (Saadnajmi) [inline packages/experimental/Icon/src/Icon.tsx:111]: Maybe add a comment linking to our issue #728 ?

**Review by Saadnajmi — APPROVED** [review]

File: packages/experimental/Icon/src/Icon.tsx [hunk 2]
```diff
@@ -51,18 +51,13 @@ function renderFontIcon(iconProps: IconProps) {
 
 function renderSvg(iconProps: IconProps) {
   const svgIconProps: SvgIconProps = iconProps.svgSource;
-  const { width, height } = iconProps;
+  const { width, height, color } = iconProps;
```
Comment (ValentinaKozlova) [inline packages/experimental/Icon/src/Icon.tsx:54]: I'll need to revert my changes because I also fixed it but in a different way :)

**Review by ValentinaKozlova — APPROVED** [review]

**Comment by Saadnajmi** [comment]
Would this close https://github.com/microsoft/fluentui-react-native/issues/888 ?
Assuing we're tracking the ColorValue in SVG issue separately in #728

File: packages/experimental/Icon/src/Icon.tsx (hunk 2 above)
Comment (rurikoaraki) [reply packages/experimental/Icon/src/Icon.tsx:54]: Ah sorry, I missed that 😞

**Comment by rurikoaraki** [comment]
@Saadnajmi Yeah it would close #888. I'll edit the comments.

File: packages/experimental/Icon/src/Icon.tsx (hunk 1 above)
Comment (rurikoaraki) [reply packages/experimental/Icon/src/Icon.tsx:112]: Done.

File: packages/experimental/Icon/src/Icon.tsx (hunk 2 above)
Comment (ValentinaKozlova) [reply packages/experimental/Icon/src/Icon.tsx:54]: It's ok ;)

**Pull request merged**